packages/jsActions/mobile-resources-native/src/permissions/RequestGenericPermission.ts

```diff
@@ -0,0 +1,65 @@
+ * @returns {Promise.<"NanoflowCommons.Enum_PermissionStatus.unavailable"|"NanoflowCommons.Enum_PermissionStatus.denied"|"NanoflowCommons.Enum_PermissionStatus.limited"|"NanoflowCommons.Enum_PermissionStatus.granted"|"NanoflowCommons.Enum_PermissionStatus.blocked">}
+ */
+export async function RequestGenericPermission(
+    permission: string
+): Promise<"unavailable" | "blocked" | "denied" | "granted" | "limited"> {
+    // BEGIN USER CODE
+    if (!permission) {
+        return Promise.reject(new Error("Input parameter 'permission' is required"));
+    }
+    const mappedPermissionName = mapPermissionName(permission);
+    if (!mappedPermissionName) {
+        console.error(`${permission} permission is not found`);
+        return Promise.resolve("unavailable");
+    }
+    const permissionStatus = await check(mappedPermissionName!);
+
+    switch (permissionStatus) {
+        case RESULTS.GRANTED:
+        case RESULTS.UNAVAILABLE:
+        case RESULTS.LIMITED:
+            return permissionStatus;
+        case RESULTS.BLOCKED:
+            handleBlockedPermission(permission);
+            return RESULTS.BLOCKED;
+        case RESULTS.DENIED:
+            return request(mappedPermissionName!);
```

Comment:
Why did you use non-null assertions for the `mappedPermissionName`? You already check if the `mappedPermissionName` variable is defined in line 47.

Reply:
done

Reply:
There is another non-null assertion on the line 61

Reply:
done